packages/optimizer/lib/transformers/AmpScriptCsp.js

```diff
@@ -0,0 +1,89 @@
+/**
+ * Copyright 2019 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+'use strict';
+
+const {calculateHash} = require('@ampproject/toolbox-script-csp');
+
+/**
+ * AmpScriptCsp - adds CSP for amp-script.
+ *
+ * Currently only supports inline scripts.
+ *
+ * This transformer supports the following parameter:
+ *
+ * `append`: specifies the calculated CSPs should be appended to existing ones.
+ */
+class AmpScriptCsp {
+  transform(tree, params) {
+    const html = tree.root.firstChildByTag('html');
+    const head = html.firstChildByTag('head');
+    const body = html.firstChildByTag('body');
+
+    if (!head || !body) return;
+
+    let hashes = new Set();
+
+    const cspMeta = this._findOrCreateCspMeta(tree, head);
+    if (params.append) {
```

Comment:
Three things: 

1. the `append` param should be config instead which is passed via the constructor (see [here](https://github.com/ampproject/amp-toolbox/blob/master/packages/optimizer/lib/transformers/AddBlurryImagePlaceholders.js#L74) for an example). I think this should be configured on a global level and not on a per-page level. 
2. can you rename the parameter to `amp-script-csp=[append|replace]`
3. please document the parameter in the optimizer readme